synapse/federation/sender/per_destination_queue.py

```diff
@@ -285,6 +323,28 @@ def _transaction_transmission_loop(self):
             # We want to be *very* sure we clear this after we stop processing
             self.transmission_loop_running = False
 
+    def _get_rr_edus(self, force_flush):
+        if not self._pending_rrs:
+            return
+        if not force_flush and not self._rrs_pending_flush:
+            # not yet time for this lot
+            return
+
+        edu = Edu(
+            origin=self._server_name,
+            destination=self._destination,
+            edu_type="m.receipt",
+            content=self._pending_rrs,
```

Comment:
I'm a bit nervous about not limiting the number of receipts going in here tbh. Can we limit it to less than 100 at least?

Reply:
I guess, though given it's a 3-level dict, that's not completely trivial. What are your concerns?

Reply:
My concern is that we end up sending 100s of read receipts in a single transaction, causing the remote to fall over. It feels a bit weird to bother limiting EDUs if we end up circumventing that for read receipts.

Reply:
If the far end falls over because it has received too many RRs, we ought to spec a limit rather than trusting other servers not to DoS us.

I'd rather hope that we won't end up with that many RRs queued up. But ok, suppose we decide to limit the number of RRs in an EDU. Presumably you'd actually want me to limit the number in a transaction rather than just sending two edus with 100 RRs in each in a single transaction?

It just feels like a bunch of work that I'm not convinced is justified. It's certainly not a nit.

Reply:
IRL conversation: it's not entirely clear what would be the right thing to do here. Let's just get it shipped